go/pkg/reader/reader.go

```diff
@@ -187,13 +188,12 @@ func NewCompressedSeeker(fs ReadSeeker) (ReadSeeker, error) {
 		fs:    fs,
 		encdW: encdW,
 		buf:   sb,
-		err:   nil,
 	}, nil
 }
 
 func (cfs *compressedSeeker) Read(p []byte) (int, error) {
-	if cfs.err != nil {
-		return 0, cfs.err
+	if !cfs.IsInitialized() {
+		return 0, errNotInitialized
 	}
 
 	var n int
```

Comment:
(I can't seem to anchor it below)

The check for `errR == nil && errW == nil` on line 201 is always true.

Reply:
Are you  sure? It's in the for loop condition, so in theory `errR` or `errW` could become non-nil during one iteration and this condition would trigger at the next iteration. Maybe it wouldn't happen in practice if the only possible error is an `io.EOF`, in which case `cfs.buf.Len() < len(p)` would be false and terminate the loop, but even in that case I think it's good to include this check in the loop condition for completeness.

Reply:
Ha, I completely missed that this was `for` rather than `if`.